design/ceph/ceph-orchestrator-osd-add-remove.md

```diff
@@ -0,0 +1,169 @@
+# Adding and removing Ceph osds via Ceph orchestrator interface
+
+## Goals
+
+1. Give Ceph's Rook Orchestrator Plugin the ability to add and remove OSDs to/from a Rook-Ceph
+   cluster running in Kubernetes.
+2. Users of the Ceph Orchestrator CLI and/or Ceph Dashboard should be able to
+   create OSDs on nodes with as much complexity as Ceph is able.
+3. Do not introduce unnecessary complexity into Rook or into Rook's CephCluster CRD.
+
+## Proposal
+
+### Removing OSDs
+Removing OSDs is the simplest place to start with this design. When removing an OSD or OSDs via the
+Rook Orchestrator Module, the Ceph mgr running in Kubernetes will run a Kubernetes Job that will run
+the required `ceph-volume` command necessary to remove the OSD(s) from a node as desired by the Ceph
+user. The Orchestrator Module will also perform whatever Ceph operations are necessary to remove the
+OSD as would be expected by the user. This will replace the manual, imperative steps currently
+required by Rook-Ceph users to remove OSDs from the storage cluster.
+
+#### OSD removal in the CephCluster resource
+The CephCluster CRD allows adding disks to nodes in several places including disks on all nodes in
+the Kubernetes cluster, to a specific node, or via disk pattern matching (in two different ways). To
+avoid the complexity of having to manage the logic for determining how to remove a node from the
+CephCluster resource, OSD removal will **not** alter the CephCluster CRD. While doing so would be
+convenient for users, there are a lot of opportunities in development to miss corner cases or
+introduce logic bugs. This may mean that Rook could re-add the disk after the disk is removed via
+the Ceph Orchestrator interface if the CephCluster CRD is configured to do so, and the user should
+be made aware of this possibility via documentation.
+
+### Adding OSDs
```

Comment:
There are two steps to adding OSDs: provisioning and starting the daemon. By running c-v we are only accomplishing the first step. Are you expecting to trigger an orchestration for the operator to complete the second part?

Reply:
I'm expecting that the discover daemon will continue to watch udev for changes to disks and trigger an orchestration when `ceph-volume` makes changes on a node. Or is this not how this process works?

Reply:
The discover daemon will watch the events, but at least with the default filter the orchestration will be skipped if it was a result of creating an OSD.